### PR TITLE
feat: improve focus outline styles for text/select inputs

### PR DESCRIPTION
**Pull request by Kyzyl-ool**
closes https://github.com/gravity-ui/uikit/issues/1043

### CODE REVIEW

File: src/components/controls/TextArea/README.md [hunk 1]
```diff
@@ -217,13 +217,14 @@ LANDING_BLOCK-->
...
+| :---------------------------------- | :----------------------- |
+| `--g-text-area-text-color`          | Text color               |
+| `--g-text-area-placeholder-color`   | Placeholder color        |
+| `--g-text-area-background-color`    | Background color         |
+| `--g-text-area-border-radius`       | Border radius            |
+| `--g-text-area-border-width`        | Border width             |
+| `--g-text-area-border-color`        | Border color             |
+| `--g-text-area-border-color-hover`  | Border color if hovered  |
+| `--g-text-area-border-color-active` | Border color if active   |
+| `--g-text-area-outline-color-focus` | Outline color if focused |
```
Comment (amje) [inline src/components/controls/TextArea/README.md:230]: `--g-text-area-focus-outline-color`

File: src/components/controls/TextInput/README.md [hunk 2]
```diff
@@ -271,14 +271,15 @@ LANDING_BLOCK-->
...
+| `--g-text-input-text-color`          | Text color               |
+| `--g-text-input-label-color`         | Label color              |
+| `--g-text-input-placeholder-color`   | Placeholder color        |
+| `--g-text-input-background-color`    | Background color         |
+| `--g-text-input-border-radius`       | Border radius            |
+| `--g-text-input-border-width`        | Border width             |
+| `--g-text-input-border-color`        | Border color             |
+| `--g-text-input-border-color-hover`  | Border color if hovered  |
+| `--g-text-input-border-color-active` | Border color if active   |
+| `--g-text-input-outline-color-focus` | Outline color if focused |
```
Comment (amje) [inline src/components/controls/TextInput/README.md:285]: `--g-text-input-focus-outline-color`

**Comment by amje** [comment]
@Kyzyl-ool Currently I see no difference, is it a bug?

**Comment by Kyzyl-ool** [comment]
> @Kyzyl-ool Currently I see no difference, is it a bug or there is no default?

I left outline color empty by default, so yes – there shouldn't be any differences. Outline on focus for inputs and selects are debatable

**Comment by Kyzyl-ool** [comment]
@amje , I would like to propose using error state's color for outline color when error state is on. Just added to the previous commit: https://github.com/gravity-ui/uikit/pull/1318/commits/48f762cdc16bf357d7006738e986763a1d96cfdd

File: src/components/Select/components/SelectControl/SelectControl.scss [hunk 3]
```diff
@@ -168,6 +168,7 @@ $blockButton: '.#{variables.$ns}select-control__button';
 
         &_error::before {
             border-color: var(--g-color-line-danger);
+            --g-select-focus-outline-color: var(--g-color-line-danger);
```
Comment (amje) [inline src/components/Select/components/SelectControl/SelectControl.scss:171]: This wont allow override this variable, we must use private/public pair here

File: src/components/controls/TextArea/TextArea.scss [hunk 4]
```diff
@@ -187,6 +189,10 @@ $block: '.#{variables.$ns}text-area';
                     &:focus-within {
                         border-color: var(--g-color-line-danger);
                     }
+
+                    &:focus-within {
+                        --g-text-area-focus-outline-color: var(--g-color-line-danger);
```
Comment (amje) [inline src/components/controls/TextArea/TextArea.scss:194]: Same as above

File: src/components/controls/TextInput/TextInput.scss [hunk 5]
```diff
@@ -380,6 +382,10 @@ $block: '.#{variables.$ns}text-input';
                     &:focus-within {
                         border-color: var(--g-color-line-danger);
                     }
+
+                    &:focus-within {
+                        --g-text-input-focus-outline-color: var(--g-color-line-danger);
```
Comment (amje) [inline src/components/controls/TextInput/TextInput.scss:387]: Same as above

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]

**Review by korvin89 — APPROVED** [review]

**Pull request merged**